### ISSUE TITLE
Investigate ability to split ngrx-immer into: ngrx-immer-signals, ngrx-immer-store, ngrx-immer-component-store

### ISSUE THREAD

**Issue by HronoSF** [timdeschryver/ngrx-immer#25]
First of all, thank you for your great work!

I have a question regarding the usage of your library. What if I only require the signals and store functionalities and do not want to install "@ngrx/component-store"? It's not necessary for my project and won't be utilized. I prefer to keep my dependencies clean.
For now I getting error:
```
Could not resolve "@ngrx/component-store"

node_modules/ngrx-immer/fesm2022/ngrx-immer-component-stone.mjs:3:54:
3 | import { ComponentStore, provideComponentStore} from '@ngrx/component-store';
```

Could you please advise me on how to achieve this with the current version of your library? If it's not possible at the moment, could we consider splitting the library into three smaller ones to match directly to my requirements?

**Comment by timdeschryver** [comment]
Do you have a reproduction @HronoSF ?
The dependencies should be optional.

**Comment by HronoSF** [comment]
@timdeschryver sure, will provide later today

**Comment by HronoSF** [comment]
@timdeschryver I've attempted to reproduce the issue on a fresh project, and it appears to work correctly with only @ngrx/signals installed. 

Upon further investigation, I've found out that the problem appeared when I've added @angular-architects/native-federation package. More detailed description of the issue I faced is in the following link:

https://www.npmjs.com/package/@angular-architects/native-federation#i-get-an-error-when-preparing-shared-packages-what-to-do

So thanks again for your project!
